### PR TITLE
Separate Requirement Files

### PR DESCRIPTION
**Pull request by Segelzwerg**
This should speed up <del>lint workflows where dependencies are not needed and</del> test workflows where linting is not required.

### CODE REVIEW

**Review by iTitus — COMMENTED** [review]
Can you keep all the files in the top-level dir?

**Comment by Segelzwerg** [comment]
> Can you keep all the files in the top-level dir?

Yes we could, but it will get messy if we think about future workflows.

**Comment by Segelzwerg** [comment]
> > Can you keep all the files in the top-level dir?
> 
> Yes we could, but it will get messy if we think about future workflows.

The idea is from here: https://github.com/pallets/flask/tree/main/requirements

**Review by iTitus — APPROVED** [review]

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -19,13 +19,14 @@ jobs:
       - uses: actions/cache@v2
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements/lint.txt') }}
```
Comment (iTitus) [inline .github/workflows/pylint.yml:22]:
```suggestion
          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements/*.txt') }}
```

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -24,16 +24,13 @@ jobs:
       - uses: actions/cache@v2
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements/requirements.txt') }}
```
Comment (iTitus) [inline .github/workflows/python-app.yml:27]:
```suggestion
          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements/*.txt') }}
```

**Review by iTitus — CHANGES_REQUESTED** [review]

File: .github/workflows/pylint.yml (hunk 1 above)
Comment (iTitus) [reply .github/workflows/pylint.yml:22]: unify caches

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (Segelzwerg) [reply .github/workflows/python-app.yml:27]: I think we shouldn't do this here because we don't want to install the linting packages in testing.

**Review by iTitus — COMMENTED** [review]
does dependabot recognize the dep files if they are not called `requirements.txt`?

**Comment by Segelzwerg** [comment]
Well I think so, but let me check again.

**Comment by Segelzwerg** [comment]
> Well I think so, but let me check again.

https://github.com/dependabot/dependabot-core/issues/1200

I guess it will work

**Review by iTitus — APPROVED** [review]

**Pull request merged**